cloudformation/curprocessor-sam.yml

```diff
@@ -2,17 +2,147 @@ AWSTemplateFormatVersion : '2010-09-09'
+
+  CostUsageReport:
+    Type: AWS::S3::Bucket
+    Properties: 
+      AccessControl: Private
+      BucketName: !Ref BucketName
+      
+  CostUsageReportBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties: 
+      Bucket: !Ref CostUsageReport
+      PolicyDocument: 
+        Id: AllowLogDelivery
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal: 
+              AWS: "arn:aws:iam::386209384616:root" 
+            Resource: !Sub ${CostUsageReport.Arn}
+            Action:
+              - s3:GetBucketAcl
+              - s3:GetBucketPolicy
+          - Effect: Allow
+            Principal: 
+              AWS: "arn:aws:iam::386209384616:root" 
+            Resource: 
+              - !Sub ${CostUsageReport.Arn}/${ReportPathPrefix}*
+              - !Sub ${CostUsageReport.Arn}/aws-programmatic-access-test-object
+            Action: s3:PutObject
+     
+  StepFunctionsRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRole
+            Principal:
+              Service: !Sub states.${AWS::Region}.amazonaws.com
+      Policies:
+        - PolicyName: InvokeLambdaFunctions
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Resource:
+                  - !GetAtt ProcessCostUsageFunction.Arn
+                  - !GetAtt CreateAthenaResourcesFunction.Arn
+                  - !GetAtt UpdateMetadataFunction.Arn
+                  - !GetAtt InitializeAthenaQueriesFunction.Arn
+                Action:
+                  - lambda:InvokeFunction
+
+  StepFunctions:
+    Type: AWS::StepFunctions::StateMachine
+    Properties:
+      StateMachineName: AWSCostUsageReport 
+      RoleArn: !GetAtt StepFunctionsRole.Arn
+      DefinitionString: !Sub |
+          {
+            "Comment": "Process AWS Cost and Usage reports and prepare them for Athena.",
+            "StartAt": "ProcessCostUsageReport",
+            "States": {
+              "ProcessCostUsageReport": {
+                "Type": "Task",
+                "Resource": "${ProcessCostUsageFunction.Arn}",
+                "Next": "CreateAthenaResources"
+              },
+              "CreateAthenaResources": {
+                "Type": "Task",
+                "Resource": "${CreateAthenaResourcesFunction.Arn}",
+                "Next": "UpdateMetadata"
+              },
+              "UpdateMetadata": {
+                "Type" : "Task",
+                "Resource": "${UpdateMetadataFunction.Arn}",
+                "Next": "InitializeAthenaQueries"
+              },
+              "InitializeAthenaQueries": {
+                "Type" : "Task",
+                "Resource": "${InitializeAthenaQueriesFunction.Arn}",
+                "End": true
+              }
+            }
+          }
+
+  xAcctStepFunctionStarterLogGroup:
```

Comment:
What is the reason for creating a Log Group in the template? Is it in order to have control over the retention period? Lambda creates log groups automatically, so other than the retention period there doesn't seem to be any benefit in defining the log group here.